cpp/source/deal.II/new_supercapacitor.templates.h

```diff
@@ -0,0 +1,358 @@
+  verbose_lvl   = solver_database.get<unsigned int>("verbosity", 0);
+
+  // distribute degrees of freedom
+  fe          = std::make_shared<dealii::FESystem<dim>>(dealii::FE_Q<dim>(1), 2);
+  dof_handler = std::make_shared<dealii::DoFHandler<dim>>(*triangulation);
+  dof_handler->distribute_dofs(*fe);
+
+  // Renumber the degrees of freedom component-wise.
+  dealii::DoFRenumbering::component_wise(*dof_handler);
+  unsigned int const n_components =
+      dealii::DoFTools::n_components(*dof_handler);
+  std::vector<dealii::types::global_dof_index> dofs_per_component(n_components);
+  dealii::DoFTools::count_dofs_per_component(*dof_handler, dofs_per_component);
+
+  // clang-format off
+  unsigned int const temperature_component      = database.get<unsigned int>("temperature_component");
+  unsigned int const solid_potential_component  = database.get<unsigned int>("solid_potential_component");
+  unsigned int const liquid_potential_component = database.get<unsigned int>("liquid_potential_component");
+  unsigned int const thermal_block              = database.get<unsigned int>("thermal_block");
+  unsigned int const electrochemical_block      = database.get<unsigned int>("electrochemical_block");
+  unsigned int const n_blocks                   = database.get<unsigned int>("n_blocks");
+  // clang-format on
+  // For now, we have just one block so we can ignore some variable.
+  std::ignore = temperature_component;
+  std::ignore = solid_potential_component;
+  std::ignore = liquid_potential_component;
+  std::ignore = thermal_block;
+  std::ignore = electrochemical_block;
+  std::ignore = n_blocks;
+
+  // read material properties
+  std::shared_ptr<boost::property_tree::ptree> material_properties_database =
+      std::make_shared<boost::property_tree::ptree>(
+          database.get_child("material_properties"));
+  MPValuesParameters<dim> params(material_properties_database);
+  params.geometry = geometry;
+  std::shared_ptr<MPValues<dim>> mp_values =
+      std::make_shared<MPValues<dim>>(params);
+
+  std::shared_ptr<boost::property_tree::ptree> boundary_values_database =
+      std::make_shared<boost::property_tree::ptree>(
+          database.get_child("boundary_values"));
+  std::shared_ptr<SuperCapacitorBoundaryValues<dim>> boundary_values =
+      std::make_shared<SuperCapacitorBoundaryValues<dim>>(
+          SuperCapacitorBoundaryValuesParameters<dim>(
+              boundary_values_database));
+
+  // Initialize the electrochemical physics parameters
+  electrochemical_physics_params.reset(
+      new ElectrochemicalPhysicsParameters<dim>(
+          std::make_shared<boost::property_tree::ptree>(database)));
+  electrochemical_physics_params->dof_handler = dof_handler;
+  electrochemical_physics_params->mp_values =
+      std::dynamic_pointer_cast<MPValues<dim> const>(mp_values);
+  electrochemical_physics_params->boundary_values =
+      std::dynamic_pointer_cast<BoundaryValues<dim> const>(boundary_values);
```

Comment:
Do you even use the ``boundary_values`` anywhere? This is meant to be deprecated. Possibly the post processor need to be updated before we can actually remove it.